seq2science/scripts/pytxi.py

```diff
@@ -0,0 +1,51 @@
+"""
+convert TPM to gene counts
+"""
+import os
+import contextlib
+
+from genomepy import Annotation
+import pandas as pd
+import pytxi
+
+
+logfile = snakemake.log[0]
+fnames = snakemake.input.cts
+assembly = snakemake.wildcards.assembly
+species = snakemake.input.fa[0]
+from_gtf = snakemake.params.from_gtf
+output = snakemake.output[0]
+sample_names = snakemake.params.names
+
+# redirect all messages to a logfile
+open(logfile, "w")  # start a clean log
+with open(logfile, "a") as log:  # appending because we mix streams (tqdm, logger & std)
```

Comment:
Can't you just do `with open(logfile, "w") as log`? Not sure what happens if you open the file with 'w' and not close it

Reply:
nope, that caused issues for some reason. Loguru should be outputting to stderr, but its clearly more complex.